src/benches/dispatch.rs

```diff
@@ -0,0 +1,147 @@
+use crate::benches::{random_bytes, random_string};
+use crate::dispatch::*;
+use crate::kv::idbstore::IdbStore;
+use crate::kv::Store;
+use wasm_bench::*;
+
+async fn opendb() -> String {
+    let dbname = random_string(12);
+    dispatch(dbname.clone(), "open".into(), "".into())
+        .await
+        .unwrap();
+    dbname
+}
+
+async fn eval(code: &str) {
+    wasm_bindgen_futures::JsFuture::from(js_sys::Promise::resolve(&js_sys::eval(code).unwrap()))
+        .await
+        .unwrap();
+}
+
+#[wasm_bench]
+async fn has(b: &mut Bench) {
+    let dbname = opendb().await;
+
+    let store = IdbStore::new(&dbname).await.unwrap().unwrap();
+    let wt = store.write().await.unwrap();
+    for i in 0..b.iterations() {
+        if i % 2 == 0 {
+            wt.put(&format!("{}", i), &random_bytes(512)).await.unwrap();
+        }
+    }
+    wt.commit().await.unwrap();
+
+    b.reset_timer();
+    eval(&format!(
```

Comment:
Little concerned about the use of `eval` (assuming that's what's happening under the covers here) deoptimizing.

Reply:
Very interested to hear more. What do you mean by that? And yes, this is using JS's eval to construct a benchmark of how long these functions take from a JS context. I should create a parallel copy of some of these that executes solely in wasm to isolate the difference that comes from the JS->wasm hop, but I'll wait for more feedback first.

Reply:
There are a lot of things js can do to accidentally trigger the jit to stop optimizing. i think eval might be one of those things. I don't know a lot about it, just to be wary of it. I'm not sure if it would just de-optimize this loop, or potentially what it's calling into like the marshaling code. @arv might know more.

Reply:
Basically I don't know if using eval() is a good simulation of somebody using the js library, but I know enough to know it's something to look at.

I guess you could find out the answer by running this in a browser and looking in the web inspector at the profiler. I know it marks functions that have been depotimized:

https://github.com/GoogleChrome/devtools-docs/issues/53

Reply:
This `eval` looks fine since it is only used once at the top level of the computation.

Reply:
But if there is any kind of iteration of `has` then this is not going to be a realistic benchmark.